file_optical.c

```diff
@@ -168,10 +168,7 @@ static void fbo_handler_init(struct fbo_handler *h, struct tcmu_device *dev,
 
 static void fbo_handler_destroy(struct fbo_handler *h)
 {
-	if (h->thr) {
-		pthread_kill(h->thr, SIGINT);
-		pthread_join(h->thr, NULL);
-	}
+	tcmulib_cancel_thread(h->thr);
```

Comment:
It looks like you handled the cases correct.  I have also tested with this.

Tested-by: Steven Royer seroyer@linux.vnet.ibm.com